File: Go/math/Areas3D.go
// Surface Area of a Cone

package main

import ("fmt" ; "math")

var Surface_Area float64

func main(){
            var r float64 = 3
            var l float64 = 7
            Surface_Area = (math.Pi * r * r) + (math.Pi * r * l)
            fmt.Print(Surface_Area)
           }


// Surface Area of a Cube

package main

import "fmt"

var Surface_Area float64

func main(){
            var a float64 = 3
            Surface_Area = 6 * a * a
            fmt.Print(Surface_Area)
           }


// Surface Area of a Cuboid

package main

import "fmt"

var Surface_Area float64

func main(){
            var a float64 = 3
            var b float64 = 5
            var c float64 = 7
            Surface_Area = 2 * ((a*b)+(b*c)+(c*a)) 
            fmt.Print(Surface_Area)
           }


// Surface Area of a Cylinder

package main

import ("fmt" ; "math")

var Surface_Area float64

func main(){
            var r float64 = 3
            var h float64 = 5
            Surface_Area = (2 * (math.Pi * r * r)) + (r*h)
            fmt.Print(Surface_Area)
           }


// Surface Area of a Pyramid

package main

import ("fmt" ; "math")

var Surface_Area float64

func main(){
            var l float64 = 3
            var b float64 = 4
            var h float64 = 5
            Surface_Area = (l*b) + (l*math.Sqrt((b/2*b/2)+(h*h))) + (b*math.Sqrt((l/2*l/2)+(h*h)))
            fmt.Print(Surface_Area)
           }
           
           
// Surface Area of a Rectangle Prism

package main

import "fmt"

var Surface_Area float64

func main(){
            var l float64 = 3
            var b float64 = 4
            var h float64 = 5
            Surface_Area = 2 * (l*b + l*h + b*h)
            fmt.Print(Surface_Area)
           }


// Time Complexity - O(n)
// Space Complexity - O(n)
           
           
// Surface Area of a Sphere
 
package main

import ("fmt" ; "math")

var Surface_Area float64

func main(){
            var r float64 = 3.5
            Surface_Area = 4 * math.Pi * r * r
            fmt.Print(Surface_Area)
           }
           
           
// Surface Area of Hemisphere

package main

import ("fmt" ; "math")

var Surface_Area float64

func main(){
            var r float64 = 3.5
            Surface_Area = (2 * math.Pi * r * r) + (math.Pi * r * r)
            fmt.Print(Surface_Area)
           }

File: Kotlin/Maths/Areas3D.kt
package kotlin.math

// Area of a Cube

fun cube(a: Double): Double {
    return 6 * a * a
}

// Area of a Cuboid

fun cuboid(l: Double, b: Double, h: Double): Double{
    return 2 * ((l*b) + (b*h) + (h*l))
}

// Area of a Cone

fun cone(r: Double, l: Double): Double{
    return (PI * r * r) + (PI * r * l)
}

// Area of a Cylinder

fun cylinder(r: Double, h: Double): Double{
    return (2.0 * (PI * r * r)) + (r * h)
}

// Area of a Sphere

fun sphere(r: Double): Double{
    return 4.0 * PI * r * r
}
// Area of a Hemi Sphere

fun hemisphere(r: Double): Double{
    return (2.0 * PI * r * r) + (PI * r * r)
}


fun main() {
    println(cube(2.0))
    println(cuboid(2.0, 4.6, 3.5))
    println(cone(5.0, 4.0))
    println(cylinder(2.0,4.0))
    println(sphere(4.5))
    println(hemisphere(2.5))
}


// Time Complexity - O(n)
// Space Complexity - O(n)
